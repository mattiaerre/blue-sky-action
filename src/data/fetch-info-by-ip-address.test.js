// https://timezoneapi.io/api/ip/?ip=66.220.144.0

require('dotenv').config();
const fetch = require('node-fetch');

const scenarios = [
  '2601:646:103:3170:e5e4:1b28:f1cc:6e74',
  '204.148.101.50',
  '185.86.151.11',
  '219.93.183.103',
  '73.202.174.69',
  '174.215.17.217'
];

describe('fetch info by ip address', () => {
  scenarios.map(scenario =>
    test(scenario, async () => {
      const url = `${process.env.TIMEZONEAPI_BASE_URL}/api/ip/?ip=${scenario}`;
      const timezone = await fetch(url)
        .then(response => response.json())
        .then(({ data }) => data);

      expect(timezone).toMatchSnapshot();
    })
  );
});
