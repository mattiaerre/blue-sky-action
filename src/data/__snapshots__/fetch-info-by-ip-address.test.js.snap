// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch info by ip address 185.86.151.11 1`] = `
Object {
  "city": "London",
  "country": "United Kingdom",
  "country_code": "GB",
  "datetime": Object {
    "date": "01/16/2018",
    "date_time": "01/16/2018 04:28:53",
    "date_time_txt": "Tuesday, January 16, 2018 04:28:53",
    "date_time_wti": "Tue, 16 Jan 2018 04:28:53 +0000",
    "date_time_ymd": "2018-01-16T04:28:53+00:00",
    "day": "16",
    "day_abbr": "Tue",
    "day_full": "Tuesday",
    "day_wilz": "16",
    "dst": "false",
    "dst_observes": "true",
    "hour_12_wilz": "04",
    "hour_12_wolz": "4",
    "hour_24_wilz": "04",
    "hour_24_wolz": "4",
    "hour_am_pm": "am",
    "minutes": "28",
    "month": "1",
    "month_abbr": "Jan",
    "month_days": "31",
    "month_full": "January",
    "month_wilz": "01",
    "offset_gmt": "+00:00",
    "offset_hours": "0",
    "offset_minutes": "0",
    "offset_seconds": "0",
    "offset_tzab": "GMT",
    "offset_tzid": "Europe/London",
    "seconds": "53",
    "time": "04:28:53",
    "timeday_gen": "night",
    "timeday_spe": "late_night",
    "week": "03",
    "year": "2018",
    "year_abbr": "18",
  },
  "ip": "185.86.151.11",
  "location": "51.5142,-0.0931",
  "postal": "EC2V",
  "state": "England",
  "state_code": "ENG",
  "timezone": Object {
    "capital": "London",
    "continent": "EU",
    "country_code": "GB",
    "country_name": "United Kingdom of Great Britain and Northern Ireland",
    "currency_alpha_code": "GBP",
    "currency_code": "826",
    "currency_country_name": "UNITED KINGDOM",
    "currency_minor_unit": "2",
    "currency_name": "Pound Sterling",
    "ds": "GB",
    "edgar": "X0",
    "fifa": "ENG,NIR,SCO,WAL",
    "fips": "UK",
    "geoname_id": "2635167",
    "gual": "256",
    "id": "Europe/London",
    "independent": "Yes",
    "ioc": "GBR",
    "iso3166_1_alpha_2": "GB",
    "iso3166_1_alpha_3": "GBR",
    "itu": "G",
    "languages": "en-GB,cy-GB,gd",
    "location": "51.50833,-0.12528",
    "marc": "xxk",
    "phone_prefix": "44",
    "tld": ".uk",
    "un_m49_code": "826",
    "wmo": "UK",
  },
}
`;

exports[`fetch info by ip address 204.148.101.50 1`] = `
Object {
  "city": "San Francisco",
  "country": "United States",
  "country_code": "US",
  "datetime": Object {
    "date": "01/15/2018",
    "date_time": "01/15/2018 20:28:53",
    "date_time_txt": "Monday, January 15, 2018 20:28:53",
    "date_time_wti": "Mon, 15 Jan 2018 20:28:53 -0800",
    "date_time_ymd": "2018-01-15T20:28:53-08:00",
    "day": "15",
    "day_abbr": "Mon",
    "day_full": "Monday",
    "day_wilz": "15",
    "dst": "false",
    "dst_observes": "true",
    "hour_12_wilz": "08",
    "hour_12_wolz": "8",
    "hour_24_wilz": "20",
    "hour_24_wolz": "20",
    "hour_am_pm": "pm",
    "minutes": "28",
    "month": "1",
    "month_abbr": "Jan",
    "month_days": "31",
    "month_full": "January",
    "month_wilz": "01",
    "offset_gmt": "-08:00",
    "offset_hours": "-8",
    "offset_minutes": "-480",
    "offset_seconds": "-28800",
    "offset_tzab": "PST",
    "offset_tzid": "America/Los_Angeles",
    "seconds": "53",
    "time": "20:28:53",
    "timeday_gen": "evening",
    "timeday_spe": "evening",
    "week": "03",
    "year": "2018",
    "year_abbr": "18",
  },
  "ip": "204.148.101.50",
  "location": "37.7749,-122.4194",
  "postal": "94119",
  "state": "California",
  "state_code": "CA",
  "timezone": Object {
    "capital": "Washington",
    "continent": "NA",
    "country_code": "US",
    "country_name": "United States of America",
    "currency_alpha_code": "USD",
    "currency_code": "840",
    "currency_country_name": "UNITED STATES",
    "currency_minor_unit": "2",
    "currency_name": "US Dollar",
    "ds": "USA",
    "edgar": "",
    "fifa": "USA",
    "fips": "US",
    "geoname_id": "6252001",
    "gual": "259",
    "id": "America/Los_Angeles",
    "independent": "Yes",
    "ioc": "USA",
    "iso3166_1_alpha_2": "US",
    "iso3166_1_alpha_3": "USA",
    "itu": "USA",
    "languages": "en-US,es-US,haw,fr",
    "location": "34.05222,-118.24278",
    "marc": "xxu",
    "phone_prefix": "1",
    "tld": ".us",
    "un_m49_code": "840",
    "wmo": "US",
  },
}
`;

exports[`fetch info by ip address 219.93.183.103 1`] = `
Object {
  "city": "Kuala Selangor",
  "country": "Malaysia",
  "country_code": "MY",
  "datetime": Object {
    "date": "01/16/2018",
    "date_time": "01/16/2018 12:28:53",
    "date_time_txt": "Tuesday, January 16, 2018 12:28:53",
    "date_time_wti": "Tue, 16 Jan 2018 12:28:53 +0800",
    "date_time_ymd": "2018-01-16T12:28:53+08:00",
    "day": "16",
    "day_abbr": "Tue",
    "day_full": "Tuesday",
    "day_wilz": "16",
    "dst": "false",
    "dst_observes": "true",
    "hour_12_wilz": "12",
    "hour_12_wolz": "12",
    "hour_24_wilz": "12",
    "hour_24_wolz": "12",
    "hour_am_pm": "pm",
    "minutes": "28",
    "month": "1",
    "month_abbr": "Jan",
    "month_days": "31",
    "month_full": "January",
    "month_wilz": "01",
    "offset_gmt": "+08:00",
    "offset_hours": "8",
    "offset_minutes": "480",
    "offset_seconds": "28800",
    "offset_tzab": "MYT",
    "offset_tzid": "Asia/Kuala_Lumpur",
    "seconds": "53",
    "time": "12:28:53",
    "timeday_gen": "morning",
    "timeday_spe": "late_morning",
    "week": "03",
    "year": "2018",
    "year_abbr": "18",
  },
  "ip": "219.93.183.103",
  "location": "3.35,101.25",
  "postal": "45400",
  "state": "Selangor",
  "state_code": "10",
  "timezone": Object {
    "capital": "Kuala Lumpur",
    "continent": "AS",
    "country_code": "MY",
    "country_name": "Malaysia",
    "currency_alpha_code": "MYR",
    "currency_code": "458",
    "currency_country_name": "MALAYSIA",
    "currency_minor_unit": "2",
    "currency_name": "Malaysian Ringgit",
    "ds": "MAL",
    "edgar": "N8",
    "fifa": "MAS",
    "fips": "MY",
    "geoname_id": "1733045",
    "gual": "153",
    "id": "Asia/Kuala_Lumpur",
    "independent": "Yes",
    "ioc": "MAS",
    "iso3166_1_alpha_2": "MY",
    "iso3166_1_alpha_3": "MYS",
    "itu": "MLA",
    "languages": "ms-MY,en,zh,ta,te,ml,pa,th",
    "location": "3.16666,101.7",
    "marc": "my",
    "phone_prefix": "60",
    "tld": ".my",
    "un_m49_code": "458",
    "wmo": "MS",
  },
}
`;

exports[`fetch info by ip address 2601:646:103:3170:e5e4:1b28:f1cc:6e74 1`] = `
Object {
  "city": "San Francisco",
  "country": "United States",
  "country_code": "US",
  "datetime": Object {
    "date": "01/15/2018",
    "date_time": "01/15/2018 20:28:53",
    "date_time_txt": "Monday, January 15, 2018 20:28:53",
    "date_time_wti": "Mon, 15 Jan 2018 20:28:53 -0800",
    "date_time_ymd": "2018-01-15T20:28:53-08:00",
    "day": "15",
    "day_abbr": "Mon",
    "day_full": "Monday",
    "day_wilz": "15",
    "dst": "false",
    "dst_observes": "true",
    "hour_12_wilz": "08",
    "hour_12_wolz": "8",
    "hour_24_wilz": "20",
    "hour_24_wolz": "20",
    "hour_am_pm": "pm",
    "minutes": "28",
    "month": "1",
    "month_abbr": "Jan",
    "month_days": "31",
    "month_full": "January",
    "month_wilz": "01",
    "offset_gmt": "-08:00",
    "offset_hours": "-8",
    "offset_minutes": "-480",
    "offset_seconds": "-28800",
    "offset_tzab": "PST",
    "offset_tzid": "America/Los_Angeles",
    "seconds": "53",
    "time": "20:28:53",
    "timeday_gen": "evening",
    "timeday_spe": "evening",
    "week": "03",
    "year": "2018",
    "year_abbr": "18",
  },
  "ip": "2601:646:103:3170:e5e4:1b28:f1cc:6e74",
  "location": "37.7957,-122.4209",
  "postal": "94109",
  "state": "California",
  "state_code": "CA",
  "timezone": Object {
    "capital": "Washington",
    "continent": "NA",
    "country_code": "US",
    "country_name": "United States of America",
    "currency_alpha_code": "USD",
    "currency_code": "840",
    "currency_country_name": "UNITED STATES",
    "currency_minor_unit": "2",
    "currency_name": "US Dollar",
    "ds": "USA",
    "edgar": "",
    "fifa": "USA",
    "fips": "US",
    "geoname_id": "6252001",
    "gual": "259",
    "id": "America/Los_Angeles",
    "independent": "Yes",
    "ioc": "USA",
    "iso3166_1_alpha_2": "US",
    "iso3166_1_alpha_3": "USA",
    "itu": "USA",
    "languages": "en-US,es-US,haw,fr",
    "location": "34.05222,-118.24278",
    "marc": "xxu",
    "phone_prefix": "1",
    "tld": ".us",
    "un_m49_code": "840",
    "wmo": "US",
  },
}
`;
